### PR TITLE
Land current code for nullsafety.dartpad.dev into nnbd branch

### PR DESCRIPTION
**Pull request by RedBrogdon**

### CODE REVIEW

**Comment by RedBrogdon** [comment]
CC @devoncarew

File: lib/src/sdk_manager.dart [hunk 1]
```diff
@@ -61,7 +61,7 @@ class PlatformSdk extends Sdk {
   String get versionFull => _versionFull;
 
   @override
-  String get sdkPath => path.dirname(path.dirname(Platform.resolvedExecutable));
+  String get sdkPath => path.join(Directory.current.path, 'dart-sdk');
```
Comment (johnpryan) [inline lib/src/sdk_manager.dart]: What's the reason for this change? It might be subject to break if the SDK is installed somewhere else (like on my mac)?

File: test/analysis_server_test.dart [hunk 2]
```diff
@@ -61,28 +61,28 @@ void main() { foo() }
...
 void main() {
   int a = <String, int>{'a': 1}['a']!;
-  print(a == null ? 'null' : 'not null');
+  print(a);
 }
 ''';
 
 const sampleNNBDCoreLibraryFailureCode = '''
 void main() {
   int a = <String, int>{'a': 1}['a'];
```
Comment (domesticmouse) [inline test/analysis_server_test.dart]: I'm curious what the difference between `sampleNNBDCoreLibraryCode` and `sampleNNBDCoreLibraryFailureCode`

Comment (RedBrogdon) [reply test/analysis_server_test.dart]: An exclamation point.

Comment (domesticmouse) [reply test/analysis_server_test.dart]: It really hides in there, doesn't it?

Comment (RedBrogdon) [reply test/analysis_server_test.dart]: Probably a good indicator I should throw a comment on it.

Comment (domesticmouse) [reply test/analysis_server_test.dart]: Sgtm =)

**Comment by RedBrogdon** [comment]
This PR is still a throwaway. I don't intend to land it, and am keeping it here mainly as a way to compare master and the null safety changes.

**Comment by domesticmouse** [comment]
Converted to draft status to reflect the non-goal of landing this PR.

**Comment by RedBrogdon** [comment]
> Converted to draft status to reflect the non-goal of landing this PR.

Thanks. I know realize we don't have the WIP checker on this repo. We should set that up.

File: lib/src/sdk_manager.dart (hunk 1 above)
Comment (RedBrogdon) [reply lib/src/sdk_manager.dart:82]: (I'm sure we've talked about this in between now and when it was posted, but I'd still feel guilty closing out the comment without responding)

At the time this code was written, it was necessary to do a custom build of the SDK in order to get access to null safety. That's no longer true, but we still need one for *sound null safety*. 

As a result, I had to add a step to the Dockerfile that downloads my custom build of the Dart SDK from cloud storage while creating the image. This class represents that SDK (or a local SDK in the same folder, when building/testing locally).

**Comment by RedBrogdon** [comment]
@domesticmouse Mind taking another look at this before it lands?
